Add redirect links to the contributor wiki (#106863)
pytorch/pytorch: docs/source/community/contribution_guide.rst

..note::

>    This page has been deprecated. Please refer to the `Contribution Guide <https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions>`_ on the PyTorch Wiki.

Reviewer: ```suggestion
   For more guidelines, see the `Contribution Guide Wiki <https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions>`_ 
```

Author: @albanD - what are your thoughts on an explicit deprecation callout v/s only linking out to current version?

Reviewer 2: I don't think this we should share this link to get "more" information. It is intended as a replacement.
Deprecated or being moved to or outdated sounds like good phrasing.
WDYT @svekars ?

Reviewer: Is this page really deprecated? Did we do a thorough review of what's on that page and decided we will replace with what's in the wiki? If yes, sure.

Author: Yes, we did a thorough review - all of the info on this page has been updated on the wiki, and this page is not the current source of truth. Having the explicit callout will disambiguate this and prevent confusion for the user

Reviewer: Sounds good to me! Approving to unblock. @subramen do you have a plan to make further changes to the docs/source/community/contribution_guide.rst page so that we remove all irrelevant information?

Author: @svekars  No further changes expected from me. I'm not aware of the archival policy for documentation, but perhaps we could follow that.
